tests/unit/src/test/scala/tests/codeactions/AddMissingMatchCaseLspSuite.scala

```diff
@@ -0,0 +1,42 @@
+package tests.codeactions
+
+class AddMissingMatchCaseLspSuite
+    extends BaseCodeActionLspSuite("addMissingMatchCase") {
+
+  check(
+    name = "basic",
+    input = """|package a
+               |
+               |sealed trait Tree
+               |case class Node(l:Tree, r:Tree) extends Tree
+               |case class Leaf(v: String) extends Tree
+               |
+               |object A {
+               |  def method(tree: Tree) = {
+               |    <<tree>> match {
+               |      case Node(_, _) => ???
+               |    }
+               |  }
+               |}
+               |""".stripMargin,
+    expectedActions = s"""|Extract `tree` as value
+                          |Add missing cases
+                          |""".stripMargin,
+    selectedActionIndex = 1,
+    expectedCode = """|package a
+                      |
+                      |sealed trait Tree
+                      |case class Node(l:Tree, r:Tree) extends Tree
+                      |case class Leaf(v: String) extends Tree
+                      |
+                      |object A {
+                      |  def method(tree: Tree) = {
+                      |    tree match {
+                      |      case Node(_, _) => ???
+                      |      case Leaf(_) => ???
+                      |    }
+                      |  }
```

Comment:
The test fails for now because the `Leaf` case is added in wrong place. 
The current output is:
```scala
Leaf(_) match {
  case Node(_, _) => ???
}
```

My idea is to create something like `AutoImportProvider` where I would get access to the typedTree and more importantly its parent, which in our case is 
```
tree match {
  case Node(_, _) => ???
}
```
I could then use the endRange of such tree, and insert the missing cases just one line before it. 

Though maybe there is a simpler way. Let me know what you think.

Reply:
I think untyped trees should be enough here. I'm not sure what you mean by "the parent of the whole match". The whole match should be a `Term.Match`. Using that you should be able to find the last case and place new ones after it. Here you can explore the syntax trees: https://astexplorer.net.

Reply:
By "parent of the whole match" I meant that I was getting:
```scala
def method(tree: Tree) = {
   tree match {
     case Node(_, _) => ???
     case Leaf(_) => ???
   }
 }
```

but I must have had something wrong because now it works as expected.